### ISSUE TITLE
Support for [project] key in schema validation

### ISSUE THREAD

**Issue by KyleMit** [tamasfe/taplo#646]
When I have the following in my `pyproject.toml`

```ini
[project]
name = "jsonschema"

[project.urls]
Homepage = "https://github.com/python-jsonschema/jsonschema"
```

I get the following error message:

```
{"name":"jsonschema","urls":{"Homepage":"https://github.com/python-jsonschema/jsonschema"}} 
is not valid under any of the given schemas 
Even Better TOML
```

![image](https://github.com/user-attachments/assets/ad1cbd9a-39a7-4daf-b221-5bbbe6f028b0)

But this follows the python spec [Declaring project metadata: the `[project]` table]( https://packaging.python.org/en/latest/specifications/pyproject-toml/#declaring-project-metadata-the-project-table) and very [common examples](https://github.com/python-jsonschema/jsonschema/blob/611e5e791d795740ea29f460ded9aeea719b8251/pyproject.toml#L8)

Can the schema validation be updated to allow for this key?

**Comment by panekj** [comment]
What schema is currently present in your file? (You can check that in right bottom corner next to file language type)

**Comment by KyleMit** [comment]
Initially it was set to **no schema selected**

![image](https://github.com/user-attachments/assets/e02f8a1b-11bb-4622-86b2-83a9adf1564e)

But I've updatd to each of the PyProject

![image](https://github.com/user-attachments/assets/b14f61ab-6d44-480c-a76e-60af7c84a87d)

Actually, looks like it might be related to the json schema store file - https://json.schemastore.org/pyproject.json

And probably need to open an issue/PR there

https://github.com/SchemaStore/schemastore/blob/6d4407c3185f5d956a932fcce111f3f926c319f0/src/schemas/json/pyproject.json#L298-L310

**Comment by panekj** [comment]
If the schema has been set on the file (it shows up as PyProject in bottom right corner) then it's likely an issue with schema. Otherwise it could be because of https://github.com/tamasfe/taplo/issues/624

**Comment by usamoi** [comment]
It looks like a deduplicate of #316. Adding `version` field to `[project]` should work.

**Comment by abelcheung** [comment]
Yes, definitely a dup of #316. Current [`pyproject` specification](https://packaging.python.org/en/latest/specifications/pyproject-toml/#declaring-project-metadata-the-project-table) mandates either one of two below:

```toml
[project]
name = '...'
version = '...'
```

```toml
[project]
name = '...'
dynamic = ['version']
```

And [current schema](https://github.com/SchemaStore/schemastore/blob/d7b9f5dad9302bb857bb461c92caa29d8e03cf5a/src/schemas/json/pyproject.json#L433-L451) indeed follows the spec.
